linkedlist.py

```diff
@@ -0,0 +1,72 @@
+class LinkedList:
+    def __init__(self, values=None):
+        self.head = None
+        self._length = 0
+
+        if values:
+            for value in values:
+                self.head = Node(value, self.head)
+                self._length += 1
+
+    def push(self, value):
+        self.head = Node(value, self.head)
+        self._length += 1
+
+    def pop(self):
+        old_head = self.head
+        if self._length > 0:
+            self.head = self.head.next
+            self._length -= 1
+        else:
+            raise ValueError("Empty List")
+        return old_head.value
+
+    def size(self):
+        return self._length
+
+    def search(self, val):
```

Comment:
If I have `val` and can call ll.search(val)` why would I want it to return the `val`?  check the instructions.

Reply:
you are right. it should return the node not value